File: plugin/jvm/src/main/java/com/twosigma/beaker/chart/xychart/TimePlot.java
/*
 *  Copyright 2014 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.twosigma.beaker.chart.xychart;

import java.util.Date;
import java.util.List;
import java.util.TimeZone;

public class TimePlot extends XYChart {

  public XYChart setXBound(Date lower, Date upper) {
    setXBound((double) lower.getTime(), (double) upper.getTime());
    return this;
  }

  @Override
  public XYChart setXBound(List bound) {
    if (bound.size() != 2) {
      throw new IllegalArgumentException("to set the x bound, the list needs to be of size=2");
    }
    if (bound.get(0) instanceof Date && bound.get(1) instanceof Date) {
      setXBound((Date) bound.get(0), (Date) bound.get(1));
    } else {
      super.setXBound(bound);
    }
    return this;
  }

}

File: kernel/scala/src/main/scala/com/twosigma/beakerx/scala/chart/xychart/TimePlot.scala
package com.twosigma.beakerx.scala.chart.xychart

import java.util.{Date, SimpleTimeZone}

/**
  * Created by bb8 on 01/06/2017.
  */
class TimePlot extends com.twosigma.beakerx.chart.xychart.TimePlot with TimePlotProperties {

  def this(simpleTimeZone: SimpleTimeZone) {
    this()
    super.setTimeZone(simpleTimeZone)
  }
}

trait TimePlotProperties extends XYChartProperties {
  this: com.twosigma.beakerx.chart.xychart.TimePlot =>

  def xBound_=(ds: (Date, Date), dummy: Int = 0): Unit = setXBound(ds._1, ds._2)
}